src/components/Editor/Preview/index.js

```diff
@@ -47,10 +47,30 @@ class Preview extends PureComponent {
     self.onMouseDown = this.onMouseDown.bind(this);
   }
 
+  // To stop newly opened sources from inheriting the scroll-state
+  // of the currently open tab we force a scrollTo
+  topScrollNewTab(dispSource, scrollContainer) {
+    if (
+      !dispSource ||
+      !scrollContainer ||
+      dispSource.get("loadedState") == "loaded"
+    )
+      return;
+
+    try {
+      this.props.preview.result.location.line;
+    } catch (e) {
+      scrollContainer.scrollTo(0, 0);
+    }
+  }
+
```

Comment:
@lukaszsobek This try catch feels a bit hacky, can't we just default the `selectedLocation.line` to 0 if there's not a line already set? that should just work.

Reply:
That was a train of thought I had considered and also the first working version used, but setting `line` means we want to scroll to that specific line and ping it. In this case we want to get around the new tab inheriting the scroll state. While using `line: 0` should do the job it did not feel like the "right" tool to use.

The try catch is only there because the object we are testing against is nested and I wanted to avoid having to call eval or `if(a && a.b && a.b.c && a.b.c.d)`

The approach used here is actually an idea taken from https://github.com/ReactTraining/react-router/blob/master/packages/react-router-dom/docs/guides/scroll-restoration.md

Reply:
Changed it - less code, easier to read, fewer dependencies.